doc/manual/checker_library.md

```diff
@@ -85,3 +85,52 @@ traceability). The metadata produced by a check shall be included and
 described in its documentation, (e.g., if a check logs in metadata version and
 configuration of a simulator, the checker documentation shall describe the
 content of aforementioned metadata fields)._
+
+## Manifest File
+
+Each Checker Library needs to provide a manifest file with configuration
+information for the framework. The format of this file is aligned with the one
+for the [Result File (`*.xqar`)](file_formats.md#result-file-xqar) and follows
+the same specification. The only difference is that it may contain additional
+Report Modules, which are delivered as part of the Checker Library.
+
+### Example
+
+```xml
+<CheckerResults version="1.0.0">
+    <CheckerBundle name="CheckerBundle1" description="Description of the Checker Bundle" build_date="2024-03-26" version="1.0.0">
+        <Executable name="/usr/bin/python3"> <!-- apply to Linux only! Probably just "python"? -->
+            <Arg>my_fancy_script.py</Arg> <!-- Mandatory arguments, followed by the configuration.xml or input file-->
+            <Arg>--fancy-param=True</Arg>
+        </Executable>
```

Comment:
What do you mean with "location requirements"? Did you understand the diagram, that the Checker Library should be in a sub directory? IMHO not!

In the current implementation, everything needs to be in the same directory. There was no concept of different Checker Libraries. And yes, with this concept, we have at least two different installation directory:
* QC Framework
* e.g. Checker Library for ASAM OpenSCENARIO XML

And maybe
* My internal Checker Library

And that is IMHO exactly how it should work. We "only" need to solve the problem, that the framework need to find the executables/modules from the Checker Library.

That's the issue we are talking about: #11

Reply:
I would force the execution for loacations in subdirectories of the library. It is a little restrictive, but may be a security requirements (e.g., the approach for Dockerfiles).

Reply:
> What do you mean with "location requirements"? Did you understand the diagram, that the Checker Library should be in a sub directory? IMHO not!
> 
> In the current implementation, everything needs to be in the same directory. There was no concept of different Checker Libraries. And yes, with this concept, we have at least two different installation directories:
> 
> * ASAM QC Framework
> * e.g. Checker Library for ASAM OpenSCENARIO XML
> 
> And maybe
> 
> * My internal Checker Library
> 
> And that is IMHO exactly how it should work. We "only" need to solve the problem, that the framework need to find the executables/modules from the Checker Library.
> 
> That's the issue we are talking about: #11

I assumed from the diagram that at least `Checker Bundle A1.exe`, `Checker Bundle A2.exe`, and `Checker Bundle A3.exe` must be in the directory `Checker Library A Directory`. Is that wrong? Because the `contains` was quiet clear for me that these will be inside there.

However, from my point of view with the definiton of exetuables this will no longer be necessary since one executable can be at `/path/a/Checker Bundle A1.exe` and a possible second one at `/path/to/somewhere/else/Checker Bundle A2.exe`

Reply:
> would force the execution for loacations in subdirectories of the library. It is a little restrictive, but may be a security requirements (e.g., the approach for Dockerfiles).

Yeah from a security perspectives this makes probably sense. But, to be honest, if we think that programs outside the current directory can't be trusted, we probably can't trust executable files in the current directory either  :D

And when we exclude executables outside the current directory we loose all the benfits of calls like
- /usr/bin/python
- conda <something>
- /home/user/.venv/.../bin/python

and so on. So I would also allow all this.
However, to mitigate this security stuff, we should probably consider the registration again. Because, in that case we can  ensure at some point only "privileged" processes (e.g. installation processes) can place configurations (which are confirmed by the user with his installation) in the qc-frameworks manifest registration directory. So only privileged users can temper with these configurations.
This way, we can use executables outside a specific directory and also have a lower security risk (yeah I know there are also arguments against it, since the executable can also be modified - but to be honest - the installation should make sure that only privilged users can temper also these executables).